### PR TITLE
Migrate to Next.js 14 app directory

### LINKED ISSUE #152: Merge App Router changes from upstream
The [upstream `vickonrails/next-starter-peacock` repo](https://github.com/vickonrails/next-starter-peacock) converted to the Next.js App Router, so this can now be merged into our fork. (before first day of next cohort)

Doing this merge will also probably supersede this PR:

- https://github.com/upleveled/next-starter-peacock/pull/144

### PR DESCRIPTION
**Pull request by ProchaLu**
Closes #152 

This PR migrates the portfolio to Next.js 14 and the app directory structure. It imports the necessary changes from the [original portfolio](https://github.com/vickonrails/next-starter-peacock) and fixes any errors encountered during the process. 

### TODO

- [x] Pull Next Starter Peacock PR https://github.com/vickonrails/next-starter-peacock/pull/21
- [x] Add UpLeveled ESLint 
- [x] Fix all Preflight errors
- [x] Add data-test-id for Playwright test

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,4 @@
 {
-  "css.lint.unknownAtRules": "ignore"
+  "css.lint.unknownAtRules": "ignore",
+  "git.ignoreLimitWarning": true
```
Comment (karlhorky) [inline .vscode/settings.json:3]: What is this? Do we really want it?

**Review by karlhorky — CHANGES_REQUESTED** [review]
~Compare the last version of the code in this PR with the version in the vickonrails repo, and create a diff that we can use (eg. copy the code of each file to your editor and then run `git diff` and copy the output to a new comment in the PR)~

Edit: Hm, let's try something else, see below

**Review by karlhorky — CHANGES_REQUESTED** [review]
It seems like this PR doesn't have any commits from `vickonrails`, which it should have (we need to update our fork to have all of the commits that he made in https://github.com/vickonrails/next-starter-peacock)

Updating from the "upstream" (the https://github.com/vickonrails/next-starter-peacock repo) is done like this:

1. create a fresh new branch off of our `main`
2. copy the Git URL using the green Code button dropdown on https://github.com/vickonrails/next-starter-peacock
3. add a new origin eg. `git remote add vickonrails <paste URL you copied>`
4. pull in the changes `git pull vickonrails master`
5. open a new PR
6. make the necessary changes from this PR in the new PR (eg. fixing TypeScript and ESLint problems)

After you have done this, then check your branch against `vickonrails` using a "compare" link like this (change the `update-next-14` bit at the end to your new branch name):

https://github.com/vickonrails/next-starter-peacock/compare/master...upleveled:next-starter-peacock:update-next-14

**Comment by ProchaLu** [comment]
Superseded by #156

**Pull request closed without merging**